api/dataservices/interface.go

```diff
@@ -103,6 +103,7 @@ type (
 		DeleteEndpoint(ID portainer.EndpointID) error
 		GetNextIdentifier() int
 		BucketName() string
+		EndpointsByTeamID(teamID portainer.TeamID) ([]portainer.Endpoint, error)
```

Comment:
I recommend you move this function to above `Heartbeat` and below `EndpointIDByEdgeID` .
i.e. to keep related and similar functions together.